nrcatalogtools/waveform.py

```diff
@@ -1,11 +1,19 @@
 import os
-
 import h5py
 import lal
 import numpy as np
 from sxs import WaveformModes as sxs_WaveformModes
 from nrcatalogtools.lvc import compute_phi_ref, get_nr_to_lal_rotation_angles
-from . import utils
+from nrcatalogtools import utils
+from pycbc.types import TimeSeries
+from pycbc.waveform import frequency_from_polarizations
```

Comment:
Pls also move back `pycbc` imports to those functions that use them. In case a user doesn't have `pycbc` installed, they should stil be able to use other functionalities of this package (think non-data-analysis use cases).

Reply:
But pycbc is a requirement for `nrcatalogtools` which means the user will definitely have `pycbc` installed ; if that is the intended behaviour, shouldn't we remove pycbc from the requirements?

Reply:
If we are serious about making pycbc, lalsuite optional, we would want to add them as optional dependencies in build config: https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies

Reply:
+1 on this, although I am more in favour of having pycbc as a requirement.

Reply:
Thinking about this a bit more carefully, since a primary purpose of this code is to interface NR data seamlessly with data analysis packages, maybe we should keep `pycbc` as a requirement. If we want to, we can make it (and `lalsuite`) optional dependencies, but this should be done as a separate PR.

I'm marking this resolved, no need to move `pycbc` imports back to function level.